lib/src/util/__tests__/useFuzzyFilter.test.tsx

```diff
@@ -51,50 +45,129 @@ describe('useFuzzyFilter', function () {
   });
 
-  it('returns unchanged input if filter was set to null', function () {
-    const shallowRender = getShallow();
-    const [,, setFilter] = shallowRender.prop('data-result');
-    setFilter(null);
+  it('returns unchanged input if filter contains only spaces', async function() {
+    const user = userEvent.setup(),
+        view = renderWithFilter(),
+        inputEl = view.getByRole('textbox');
+    let outputItems = view.getAllByTestId('result');
+
+    expect(outputItems.length).toBe(4);
+    await user.type(inputEl, '[Space]');
 
-    const [output, filter] = shallowRender.prop('data-result');
-    expect(output).toBe(input);
-    expect(filter).toBe(null);
+    outputItems = view.getAllByTestId('result');
+    expect(outputItems.length).toBe(4);
   });
 
-  it('filters input and returns new filter term', function () {
-    const shallowRender = getShallow();
-    const [,, setFilter] = shallowRender.prop('data-result');
-    setFilter('Mo');
-
-    const [output, filter] = shallowRender.prop('data-result');
-    expect(output).toEqual([
-      {
-        id: 'motorcycle',
-        name: 'Motorcycle'
-      }, {
-        id: 'moped',
-        name: 'Moped'
-      }
-    ]);
-    expect(filter).toBe('Mo');
+  it('filters input and returns new filter term', async function() {
+    const user = userEvent.setup(),
+        view = renderWithFilter(),
+        inputEl = view.getByRole('textbox');
+    let outputItems = view.getAllByTestId('result');
+
+    expect(outputItems.length).toBe(4);
+    await user.type(inputEl, 'cy');
+
+    outputItems = view.getAllByTestId('result');
+    expect(outputItems.length).toBe(2);
+    expect(outputItems[0]).toHaveTextContent('Bicycle');
+    expect(outputItems[1]).toHaveTextContent('Motorcycle');
+    expect(inputEl).toHaveValue('cy');
+  });
+
+  it('uses trimmed filter for search', async function() {
+    const user = userEvent.setup(),
+        view = renderWithFilter(),
+        inputEl = view.getByRole('textbox');
+    let outputItems = view.getAllByTestId('result');
+
+    expect(outputItems.length).toBe(4);
+    await user.type(inputEl, ' cycle');
+
+    outputItems = view.getAllByTestId('result');
+    expect(outputItems.length).toBe(2);
+    expect(outputItems[0]).toHaveTextContent('Bicycle');
+    expect(outputItems[1]).toHaveTextContent('Motorcycle');
+    expect(inputEl).toHaveValue(' cycle');
+  });
+
+  it('matches even when the term is located at the tail of the string', async function() {
+    const user = userEvent.setup(),
+        view = renderWithFilter(),
+        inputEl = view.getByRole('textbox');
+    let outputItems = view.getAllByTestId('result');
+
+    expect(outputItems.length).toBe(4);
+    await user.type(inputEl, 'scooter');
+
+    outputItems = view.getAllByTestId('result');
+    expect(outputItems.length).toBe(1);
+    expect(outputItems[0]).toHaveTextContent('Moped or Scooter');
+    expect(inputEl).toHaveValue('scooter');
+  });
```

Comment:
I don't think this test is necessary. For an approximate string matcher, it's bound to find 'scooter' regardless of where it appears.

Reply:
> I don't think this test is necessary. For an approximate string matcher, it's bound to find 'scooter' regardless of where it appears.

I was unsure about this one as well. I used `fuzzyFilter.test.ts` as a guideline for tests, and it had this test included. Remove?

Reply:
> > I don't think this test is necessary. For an approximate string matcher, it's bound to find 'scooter' regardless of where it appears.
> 
> I was unsure about this one as well. I used `fuzzyFilter.test.ts` as a guideline for tests, and it had this test included. Remove from `useFuzzyFilter.test.ts`?

I looked a bit into this and since our fuse.js options set the `ignoreLocation` to true, I think the test should stay. https://fusejs.io/api/options.html#ignorelocation

Reply:
> I looked a bit into this and since our fuse.js options set the `ignoreLocation` to true, I think the test should stay. https://fusejs.io/api/options.html#ignorelocation

Thanks for clarifying!